### PR TITLE
changed to RST

### PR DESCRIPTION
**Pull request by Higgs32584**
created draft pull request for issue#25582, documentation

#25582

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -108,5 +108,8 @@ lib/z.lib
 #########################
 lib/matplotlib/backends/web_backend/node_modules/
 lib/matplotlib/backends/web_backend/package-lock.json
-
+Scripts
+lib/.gitignore
+.gitignore
+pyvenv.cfg
```
Comment (tacaswell) [inline .gitignore:114]: We definitely to not want to ignore the gitignore file!


I'm not sure where `Scripts` come from, but can you put the `pyvenv.cfg` in a new section (labeled "niche tooling files" or something like that)?

File: galleries/users_explain/toolkits/axes_grid.rst [hunk 2]
```diff
@@ -0,0 +1,332 @@
+.. redirect-from:: /tutorials/toolkits/axes_grid
```
Comment (tacaswell) [inline galleries/users_explain/toolkits/axes_grid.rst:1]: Should this also be a move?

**Comment by tacaswell** [comment]
Even though this is documentation, it can not be backported because it relies on the re-organization.

**Comment by Higgs32584** [comment]
Apologies! This was more so a draft for the moment. I was not planning on committing a .git ignore

**Comment by Higgs32584** [comment]
What I am trying to resolve is the fact that the three changed files download as full commented blocks. I initially was planning on just reformatting the file, but it was suggested that I change the files to .rst.

If I was going to do that where would I move it to?

**Comment by tacaswell** [comment]
github is detecting 2 renames of `.py` -> `.rst` and reporting one wholly new `.rst`.  I suspect that means you missing a `git rm <file name>` command.

**Comment by Higgs32584** [comment]
I do not know what that is the case. I did not modify the location.
https://github.com/matplotlib/matplotlib/blob/main/galleries/users_explain/toolkits/axes_grid.py

**Comment by jklymak** [comment]
When you rename a file, you should use "git mv" so git knows that is what you did versus delete a file and add a new one. https://git-scm.com/docs/git-mv

**Comment by Higgs32584** [comment]
@jklymak Ok will do, thank you!

**Comment by Higgs32584** [comment]
would this make a solid index.rst for 

C:/Users/mjhig/.vscode/bob/matplotlib/galleries/users_explain/toolkits

```
``

.. _tutorials-toolkits:

.. redirect-from:: /tutorials/toolkits

Toolkits
--------

These tutorials cover toolkits designed to extend the functionality
of Matplotlib in order to accomplish specific goals.
.
.. toctree::
    :maxdepth: 2

    axes_grid.rst
    axisartist.rst
    mplot3d.rst
    
```

File: .gitignore [hunk 3]
```diff
@@ -1,112 +0,0 @@
-#########################################
```
Comment (tacaswell) [inline .gitignore:1]: Deleting our `.gitignore` is also not something we can merge!

**Comment by Higgs32584** [comment]
I am very sorry I can redo the pull request if necessary, I am sorry about the .gitignore it was not meant to be committed, but now that the changes are staged it got really messy, I can close this pull request if you want. It was getting annoying trying to explain the code, so I made a pull request to get feedback.

**Comment by ksunden** [comment]
FYI, you can tweak your description a bit to automatically link the issue by using special codewords like "Closes #XXXXX". I have added it to this PR manually, but for future reference.

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

As for the gitignore stuff, consider rebasing, no need to start a new PR if the overall change is still wanted (I honestly haven't taken a close look), it can be done here:

https://matplotlib.org/stable/devel/development_workflow.html#rewriting-commit-history

**Comment by Higgs32584** [comment]
Can someone tell me if I put the .rst files in the right place? Also I'm wondering if the index.rst works to

**Comment by jklymak** [comment]
No we can't tell you if you have put it in the right place because you haven't committed it.  You may want to carefully go through our guide to git and Github, it is actually relatively decent: https://matplotlib.org/stable/devel/development_workflow.html. Let us know if parts of it are not clear.

**Comment by Higgs32584** [comment]
@jklymak ok I will!! I apologize, I haven’t used GitHub with this much version control before! I have only done small projects before! I really look forward to completing my contribution. Thank you!

**Comment by Higgs32584** [comment]
H

**Comment by Higgs32584** [comment]
accident

**Comment by jklymak** [comment]
@Higgs32584 This is unlikely to be building properly though for some reason CI is not running for the job.  But the rst files must be formatted like rst files.  You probably also have to `git remove` the `README.txt` from the source directory.  

You should be able to build and check this locally using `make html-skip-subdirs`

**Comment by melissawm** [comment]
Hello @Higgs32584 - let me know if you'd like to connect and come up with a plan for this ☺️ We can chat over at [gitter](https://gitter.im/matplotlib/matplotlib) or set up a meeting over zoom. Cheers!

**Comment by Higgs32584** [comment]
Ok, I will! I just did not know how to set up a circle CI pipeline

**Comment by melissawm** [comment]
I *think* this is on us @Higgs32584 - see #25672 (takeaway is don't worry about it 😄 )

**Comment by Higgs32584** [comment]
Hi, is there anything else that needs to be done? [melissawm](https://github.com/melissawm)

File: galleries/users_explain/toolkits/axes_grid.rst
Comment (jklymak) [inline galleries/users_explain/toolkits/axes_grid.rst:1]: I think these still need to strip the `"""` at the beginning and end, or it won't build.

Comment (Higgs32584) [reply galleries/users_explain/toolkits/axes_grid.rst:1]: my bad! removed quotations! Anything else I can do? Thank you for being so patient!

File: galleries/users_explain/toolkits/index.rst [hunk 4]
```diff
@@ -0,0 +1,32 @@
+.. _tutorials-toolkits:
+
+.. redirect-from:: /tutorials/toolkits
+
+User Toolkits
+=============
+
+Here you can find examples and explanations of how to use various toolkits available in Matplotlib.
+
+Axes Artist
```
Comment (jklymak) [inline galleries/users_explain/toolkits/index.rst:10]: Is there a good reason to not just have these in one TOC? 

Otherwise this looks good!

Comment (Higgs32584) [reply galleries/users_explain/toolkits/index.rst:10]: I guess that is how I got it to look the way it did before

Comment (jklymak) [reply galleries/users_explain/toolkits/index.rst:10]: Before it was a bunch of broken sphinx gallery thumbnails. https://matplotlib.org/devdocs/users/explain/toolkits/index.html  I don't think anything is gained by having a title with almost the same thing written beneath it as a one-entry table of contents.

Comment (jklymak) [reply galleries/users_explain/toolkits/index.rst:10]: https://output.circle-artifacts.com/output/job/53486b53-726d-4792-9885-525ff863a886/artifacts/0/doc/build/html/users/explain/toolkits/index.html

Comment (jklymak) [reply galleries/users_explain/toolkits/index.rst:10]: "The same ways as before" was just three broken thumbnails (https://matplotlib.org/devdocs/users/explain/toolkits/index.html).  Why do you want that?  

If you want something graphical, you could create an rst page with select images that are inside the tutorials. If you want the cards that are used for sphinx gallery, you could copy their style and use them on this rst page.  You could also perhaps do something with sphinx-design. 

If you decide to do this (which I'm not asking for, I'm just asking for you not to include three subheadings that each contain a table of contents) look at examples in the source code of where images have been included.

**Comment by Higgs32584** [comment]
They need a computer science on git conflict theory

**Comment by jklymak** [comment]
```
/home/circleci/project/doc/users/explain/toolkits/axisartist.rst:563: WARNING: Bullet list ends without a blank line; unexpected unindent.
```

**Review by jklymak — APPROVED** [review]
Thanks for working through this!

**Comment by rcomer** [comment]
Congratulations on your first PR merged into Matplotlib @Higgs32584!

**Comment by Higgs32584** [comment]
Thank you! @rcomer

**Pull request merged**